src/ccover/ccover_parse_options.cpp

```diff
@@ -0,0 +1,753 @@
+/*******************************************************************\
+
+Module: CBMC Command Line Option Processing
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+/// \file
+/// CBMC Command Line Option Processing
+
+#include "ccover_parse_options.h"
+
+#include <fstream>
+#include <cstdlib> // exit()
+#include <iostream>
+#include <memory>
+
+#include <util/string2int.h>
+#include <util/config.h>
+#include <util/unicode.h>
+#include <util/invariant.h>
+#include <util/exit_codes.h>
+
+#include <langapi/language.h>
+#include <langapi/mode.h>
+
+#include <ansi-c/c_preprocess.h>
+#include <ansi-c/ansi_c_language.h>
+#include <cpp/cpp_language.h>
+
+#include <goto-programs/convert_nondet.h>
+#include <goto-programs/initialize_goto_model.h>
+#include <goto-programs/instrument_preconditions.h>
+#include <goto-programs/goto_convert_functions.h>
+#include <goto-programs/goto_inline.h>
+#include <goto-programs/link_to_library.h>
+#include <goto-programs/loop_ids.h>
+#include <goto-programs/mm_io.h>
+#include <goto-programs/read_goto_binary.h>
+#include <goto-programs/remove_function_pointers.h>
+#include <goto-programs/remove_returns.h>
+#include <goto-programs/remove_vector.h>
+#include <goto-programs/remove_complex.h>
+#include <goto-programs/remove_asm.h>
+#include <goto-programs/remove_unused_functions.h>
+#include <goto-programs/remove_skip.h>
+#include <goto-programs/set_properties.h>
+#include <goto-programs/show_goto_functions.h>
+#include <goto-programs/show_symbol_table.h>
+#include <goto-programs/show_properties.h>
+#include <goto-programs/string_abstraction.h>
+#include <goto-programs/string_instrumentation.h>
+
+#include <goto-symex/rewrite_union.h>
+#include <goto-symex/adjust_float_expressions.h>
+
+#include <goto-instrument/full_slicer.h>
+#include <goto-instrument/nondet_static.h>
+#include <goto-instrument/cover.h>
+
+#include <pointer-analysis/add_failed_symbols.h>
+
+#include <langapi/mode.h>
+
+#include <cbmc/version.h>
+
+ccover_parse_optionst::ccover_parse_optionst(int argc, const char **argv):
+  parse_options_baset(CCOVER_OPTIONS, argc, argv),
+  messaget(ui_message_handler),
+  ui_message_handler(cmdline, "ccover " CBMC_VERSION)
+{
+}
+
+void ccover_parse_optionst::eval_verbosity()
+{
+  // this is our default verbosity
+  unsigned int v=messaget::M_STATISTICS;
+
+  if(cmdline.isset("verbosity"))
+  {
+    v=unsafe_string2unsigned(cmdline.get_value("verbosity"));
+    if(v>10)
+      v=10;
+  }
+
+  ui_message_handler.set_verbosity(v);
+}
+
+void ccover_parse_optionst::get_command_line_options(optionst &options)
```

Comment:
CBMC has a much cleaned up command-line option parsing, could this please be made at least as good?

Reply:
Just as a note: Most of this code is duplicated in at least 8 different tools, which is not ideal. I'm not saying that needs to be fixed in this PR, but it has become difficult to distinguish intended from spurious differences between the command line interfaces.